src/data/finsupp/lex.lean

```diff
@@ -0,0 +1,87 @@
+/-
+Copyright (c) 2022 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa
+-/
+import data.pi.lex
+import data.finsupp.order
+
+/-!
+# Lexicographic order on finitely supported functions
+
+This file defines the lexicographic order on `finsupp`.
+-/
+
+namespace finsupp
+
+variables {α N : Type*} [has_zero N]
+
+/-- The lexicographic relation on `α →₀ N`, where `α` is ordered by `r`,
+  and `N` is ordered by `s`. -/
+protected def lex (r : α → α → Prop) (s : N → N → Prop) (x y : α →₀ N) : Prop :=
+pi.lex r s x y
```

Comment:
```suggestion
∃ i, (∀ j, r j i → x j = y j) ∧ s (x i) (y i)
```

Reply:
I think that the nicer spelling is not defeq to the original one.

Reply:
Can you add a lemma to indicate that they are the same?